Common/PkmnSpecies.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Common
+{
+    /// <summary>
+    /// A <see cref="PkmnSpecies"/> uniquely identifies and describes a species of pokemon.
+    /// Each species has an <see cref="Id"/>, and additional pokedex information
+    /// like the <see cref="Name">name</see> or pokedex <see cref="Flavors">flavor texts</see>.
+    /// <para>
+    /// Application code should never pass around raw species id strings.
+    /// Instead, it is recommended to immediately turn every raw input into an instance of this
+    /// class using <see cref="OfId"/> or <see cref="OfIdWithPokedexData"/> and use that instead.
+    /// All equality checks safely forward to the species id.
+    /// </para>
+    /// </summary>
+    public sealed class PkmnSpecies : IComparable<PkmnSpecies>
+    {
+        private static readonly Dictionary<string, string> NoFlavors = new Dictionary<string, string>();
+
+        /// <summary>
+        /// Internal dictionary that keeps track of any registered pokedex data
+        /// by providing a lookup from species ids to species instances containing said pokedex data.
+        /// </summary>
+        private static readonly Dictionary<string, PkmnSpecies> Pokedex = new Dictionary<string, PkmnSpecies>();
```

Comment:
would prefer some type of static constructor to register pokedex IDs automatically based on a json file or smth like that

Reply:
I made it the way it is right now for multiple reasons:
- Use the species class _everywhere_, do not use a raw string _anywhere_ (except at application boundaries).
- Convenience of having the pokedex readily available in the species class.
- don't transitively make everything depend on Json.NET
- don't cause _everything_ to have a slow startup time due to parsing a large json file (which would include unit tests)

In my blob of local changes I actually do have a pokedex.json in the main executable project that gets loaded and registered at startup

Reply:
I could put that blob of text in the xmldoc I guess

Reply:
that's fair, though it does raise the question of how the pokedex will get registered